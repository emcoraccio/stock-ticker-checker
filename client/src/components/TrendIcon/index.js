import React from 'react';
import { Icon } from 'react-materialize';
import './styles.css';

const TrendIcon = ({ stockTrend }) => {

  if (stockTrend > 0) {
    return (
      <Icon className="stockTrend green-text text-accent-3">trending_up</Icon>
    )
  }
  else if (stockTrend === 0) {
    return (
      <Icon className="stockTrend grey-text">trending_flat</Icon>
    )
  }
  else {
    return (
      <Icon className="stockTrend red-text text-accent-3">trending_down</Icon>
    )
  }

}

export default TrendIcon;