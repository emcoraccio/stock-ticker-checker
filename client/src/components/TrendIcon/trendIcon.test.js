import React from 'react';
import { shallow } from 'enzyme';
import TrendIcon from './index';

it('expect to render TrendIcon component', () => {
  const mockNegStockTrend = -1;
  const mockPosStockTrend = .23;
  const mockNeutStockTrend = 0;

  expect(shallow(<TrendIcon stockTrend={mockNegStockTrend} />)).toMatchSnapshot();
  expect(shallow(<TrendIcon stockTrend={mockPosStockTrend} />)).toMatchSnapshot();
  expect(shallow(<TrendIcon stockTrend={mockNeutStockTrend} />)).toMatchSnapshot();
})