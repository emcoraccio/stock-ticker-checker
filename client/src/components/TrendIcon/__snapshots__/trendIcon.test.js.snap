// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render TrendIcon component 1`] = `ShallowWrapper {}`;

exports[`expect to render TrendIcon component 2`] = `ShallowWrapper {}`;

exports[`expect to render TrendIcon component 3`] = `ShallowWrapper {}`;
